src/btree/row_modify.c

```diff
@@ -425,22 +424,12 @@ __wt_update_obsolete_check(
      */
     if (count > WT_THOUSAND) {
         WT_STAT_CONN_INCR(session, cache_eviction_force_long_update_list);
-        __wt_page_evict_soon(session, cbt->ref);
+        __wt_page_evict_soon(session, ref);
     }
 
     if (next != NULL)
         __wt_free_update_list(session, &next);
-    else {
```

Comment:
Why this check is no longer useful? I think not doing retry checks on long list makes sense to me.

Reply:
The current skip logic of saving the `last_running` transaction id and skipping the obsolete check for other update chains until this transaction is committed is not the correct way to skipping. It is possible that the other update chains may have obsolete content with transaction ids smaller the `last_running`, but we do not know. 

So the current logic turns out to be a random skip of update chains, instead of that check for obsolete updates on all the keys when the page size is grown to a size where it may have obsolete updates would be good.

Reply:
Yes, it's wrong to skip all the checks on the entire page.

Reply:
No, we added another check for checking the page size. If the page size is greater than 50% of the split size, then only the obsolete check function is executed. This is not a perfect solution, but it is better than the current logic.

Reply:
Thanks, I understand now.